openapi/generator.go

```diff
@@ -758,6 +758,22 @@ func (g *Generator) newSchemaFromStructField(sf reflect.StructField, required bo
 	if t, ok := sf.Tag.Lookup(formatTag); ok {
 		schema.Format = t
 	}
+
+	// Set example value from tag to schema
+	if e := sf.Tag.Get("example"); strings.TrimSpace(e) != "" {
+		if parsed, err := parseExampleValue(sf.Type, e); err != nil {
+			g.error(&FieldError{
+				Message:  fmt.Sprintf("example value for field '%s' cannot be parsed: %s", e, err),
```

Comment:
the param `e` isn't the field's name, but the raw value being parsed.
Just use smthing like `example value %q cannot be parsed: %s`.